src/domain/handler.js

```diff
@@ -0,0 +1,103 @@
+import { askQuestion, rl } from "../utils/readlineAPI.js";
+import { INFO_MESSAGE } from "./constant.js";
+
+import {
+  LottoStore,
+  Machine,
+  Buyer,
+  initializeDataStorage,
+  dataStorage,
+  LottoMainCompany,
+} from "./index.js";
+
+import {
+  ViewBuyerLottoList,
+  ViewProfitRate,
+  ViewPurchaseMessage,
+  ViewWinnerPriceStats,
+} from "../UI/View/OutputView.js";
+import { startLottoGame } from "../step2-index.js";
+
+const handleRequestPayment = async () => {
+  try {
+    const purchaseAmount = await askQuestion(INFO_MESSAGE.REQUEST_PAYMENT);
+    const lottoCount = LottoStore.calculateLottoAmount(Number(purchaseAmount));
+    ViewPurchaseMessage(lottoCount);
+
+    const buyerLottoList = Buyer.generateLottoList(lottoCount);
+    ViewBuyerLottoList(buyerLottoList);
+
+    return;
+  } catch (error) {
+    console.log(error.message);
+    return handleRequestPayment();
+  }
+};
+
+const handleWinNumbers = async () => {
+  try {
+    const inputWinNumbers = await askQuestion(INFO_MESSAGE.REQUEST_WIN_NUMBER);
+    const winNumbers = Machine.getWinnerLottoList(inputWinNumbers);
+
+    return winNumbers;
+  } catch (error) {
+    console.log(error.message);
+    return handleWinNumbers();
+  }
+};
+
+const handleBonusNumbers = async () => {
+  try {
+    const inputBonusNumber = await askQuestion(
+      INFO_MESSAGE.REQUEST_BONUS_NUMBER
+    );
+    Machine.getBonusNumber(dataStorage.winNumbers, inputBonusNumber);
+  } catch (error) {
+    console.log(error.message);
+    return handleBonusNumbers();
+  }
+};
+
+const handleWinnerStats = () => {
+  try {
+    const winnerStats = LottoMainCompany.handleWinnerStats();
+    ViewWinnerPriceStats(winnerStats);
+  } catch (error) {
+    console.log(error.message);
+  }
+};
+
+const handleProfitRate = () => {
+  try {
+    const profitRate = Buyer.calculateLottoProfit();
+    ViewProfitRate(profitRate);
+  } catch (error) {
+    console.log(error.message);
+  }
+};
+
+const handleRestartOrExit = async () => {
+  try {
+    const restartOrExitLotto = await askQuestion(INFO_MESSAGE.REQUEST_RESTART);
+
+    if (restartOrExitLotto.toLowerCase() !== "y") {
+      return rl.close();
+    } else {
+      console.clear();
+      initializeDataStorage();
+      return startLottoGame();
+    }
```

Comment:
스코프를 줄인다는 것은 아래와 같은 맥락이랍니다!
`{ }` 스코프는 많이 있으면 가독성을 해치거든요.

```suggestion
    if (restartOrExitLotto.toLowerCase() !== "y") {
      return rl.close();
    }
    
    console.clear();
    initializeDataStorage();
    startLottoGame();
```